src/vi/vlogtruyen/src/eu/kanade/tachiyomi/extension/vi/vlogtruyen/VlogTruyen.kt

```diff
@@ -0,0 +1,228 @@
+        status = parseStatus(document.select("div.top-detail-manga > div.top-detail-manga-avatar > div.manga-status > p").firstOrNull()?.text())
+        author = document.select(".h5-drawer:contains(Tác Giả) + ul li a").joinToString { it.text() }
+    }
+
+    private fun parseStatus(status: String?) = when {
+        status == null -> SManga.UNKNOWN
+        status.contains("Đang tiến hành") -> SManga.ONGOING
+        status.contains("Đã hoàn thành") -> SManga.COMPLETED
+        status.contains("Tạm ngưng") -> SManga.ON_HIATUS
+        else -> SManga.UNKNOWN
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val html = response.body.string().replace("\\", "").replace("n  ", "").substringAfter("\"chaptersHtml\":\"").substringBefore("\",")
+        val document = response.asJsoup(html)
+        val hidePaidChapters = preferences.getBoolean(KEY_HIDE_PAID_CHAPTERS, false)
+        return document.select("li, .ul-list-chaper-detail-commic li").filterNot {
+            if (hidePaidChapters) {
+                it.select("li > b").text().isNotBlank().or(!hidePaidChapters)
+            } else {
+                it.select("li > a").text().isBlank().or(false)
+            }
+        }
+            .mapNotNull {
+                SChapter.create().apply {
+                    setUrlWithoutDomain(it.select("a").first()!!.attr("href"))
+                    name = it.select("h3").first()!!.text().replaceAfter("-", "").replaceAfter(":", "")
+                        .replaceAfter("0 ", "").replaceAfter("1 ", "").replaceAfter("2 ", "")
+                        .replaceAfter("3 ", "").replaceAfter("4 ", "").replaceAfter("5 ", "")
+                        .replaceAfter("6 ", "").replaceAfter("7 ", "").replaceAfter("8 ", "")
+                        .replaceAfter("9 ", "").replace("-", "").replace(":", "")
+                    if (it.select("li > b").text().isNotBlank()) {
+                        name += " " + it.select("li > b").text() + " 🔒"
+                    }
+                    date_upload = parseDate(it.select("li > span").firstOrNull()!!.text())
+                }
+            }
+    }
+
+    private fun parseDate(date: String): Long = runCatching {
+        dateFormat.parse(date)?.time
+    }.getOrNull() ?: 0L
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val url = client.newCall(
+            GET(
+                when {
+                    manga.url.startsWith("http") -> manga.url
+                    else -> "$baseUrl${manga.url}"
+                },
+            ),
+        ).execute().asJsoup()
+        if (checkChapterLists(url).isNotEmpty()) {
+            val mangaId = checkChapterLists(url)
+            return client.newCall(GET("$baseUrl/thong-tin-ca-nhan?manga_id=$mangaId", headers))
+                .asObservableSuccess()
+                .map { response -> chapterListParse(response) }
+        }
+        return super.fetchChapterList(manga)
+    }
```

Comment:
you can simple make another call with id in the `chapterListParse` method, also don't return super method when chapters not found, as it will make another network call